CONTRIBUTING.md

```diff
@@ -500,6 +500,58 @@ it is acceptable to follow the convention in that module; though in general, it
 would be preferable to follow the motto of "leave the codebase in better shape
 than you found it."
 
+## Error Codes and Messages
+
+Errors are a part of the contract between a caller and callee in the same way
+parameters and return values are. When writing an error message or selecting an
+error code, consider the perspective of the caller and their natural first
+question: "how can I make this error go away?"
+
+Good messages should be to the point and in terms of the caller and not
+artifacts of an implementation detail in the callee.
+
+Compare
+
+```cpp
+Result<void> CheckNumber(int number) {
+  if (!number) {
+    int u = n / number;
+  } else {
+    return KATANA_ERROR(IllegalArgument, "cannot divide by zero");
+  }
+}
+```
+
+and
+
+```cpp
+Result<void> CheckNumber(int number) {
+  if (!number) {
+    int u = n / number;
+  } else {
+    return KATANA_ERROR(IllegalArgument, "number should be positive");
+  }
```

Comment:
Argument is an int, so "should be non-negative"?

Reply:
maybe I was being too obscure, but I was trying to convey non-zero as well (zero being neither negative nor positive).